src/UmikoBot.cpp

```diff
@@ -289,22 +291,16 @@ UmikoBot::UmikoBot(QObject* parent)
 	}});
 
 	m_commands.push_back({ Commands::GLOBAL_SET_PREFIX, "setprefix",
-		[this](Discord::Client& client, const Discord::Message& message, const Discord::Channel& channel)
+		[this](Client& client, const Message& message, const Channel& channel)
 	{
-		Permissions::ContainsPermission(client, channel.guildId(), message.author().id(), CommandPermission::ADMIN,
-			[this, channel, message](bool result)
-		{
-			GuildSetting& s = GuildSettings::GetGuildSetting(channel.guildId());
-			if (!result)
-			{
-				createMessage(channel.id(), "You don't have permissions to use this command.");
-				return;
-			}
+		GuildSetting& s = GuildSettings::GetGuildSetting(channel.guildId());
+		QStringList args = message.content().split(' ');
 
-			QStringList args = message.content().split(' ');
-			if (args.first() != s.prefix + "setprefix")
-				return;
+		if (args.first() != s.prefix + "setprefix")
+			return;
```

Comment:
Isn't this handled in the onMessage before it calls the corresponding command?

Reply:
Ah well I supposed to remove all of those redundant conditionals, but I forgot to check the files that were added much before...